pkg/api/pod/util.go

```diff
@@ -225,3 +228,20 @@ func UpdatePodCondition(status *api.PodStatus, condition *api.PodCondition) bool
 	// Return true if one of the fields have changed.
 	return !isEqual
 }
+
+// DropDisabledAlphaFields removes disabled fields from the pod spec.
+// This should be called from PrepareForCreate/PrepareForUpdate for all resources containing a pod spec.
+func DropDisabledAlphaFields(podSpec *api.PodSpec) {
```

Comment:
Do we need to drop the resource field as well?

Reply:
no, that's a key/value in an existing field... these are new fields